collect_app/src/main/java/org/odk/collect/android/utilities/SharedPreferencesUtils.java

```diff
@@ -0,0 +1,281 @@
+
+    private final Context context = Collect.getInstance();
+    private final SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
+    private final SharedPreferences.Editor editor = sharedPrefs.edit();
+    private final SharedPreferences adminSharedPrefs =
+            Collect.getInstance().getSharedPreferences(ADMIN_PREFERENCES, MODE_PRIVATE);
+    private final SharedPreferences.Editor adminEditor = adminSharedPrefs.edit();
+
+    String getJSONFromPreferences(Collection<String> keys) throws JSONException {
+        SharedPreferencesUtils obj = new SharedPreferencesUtils();
+        keys.addAll(ALL_GENERAL_KEYS);
+        JSONObject sharedPrefJson = obj.getModifiedPrefs(keys);
+        Timber.i(sharedPrefJson.toString());
+        return sharedPrefJson.toString();
+    }
+
+    private JSONObject getModifiedPrefs(Collection<String> keys) throws JSONException {
+        JSONObject prefs = new JSONObject();
+        JSONObject adminPrefs = new JSONObject();
+        JSONObject generalPrefs = new JSONObject();
+
+        for (String key : keys) {
+            String stringValue;
+            String defaultStringValue;
+
+            // checking for admin password
+            if (key.equals(KEY_ADMIN_PW)) {
+                stringValue = adminSharedPrefs.getString(key, "");
+                if (!stringValue.equals("")) {
+                    adminPrefs.put(key, stringValue);
+
+                    //skip further checking
+                    continue;
+                }
+            }
+
+            try {
+                stringValue = getStringValue(key);
+                defaultStringValue = getDefaultStringValue(key);
+                if (!stringValue.equals(defaultStringValue)) {
+                    generalPrefs.put(key, stringValue);
+                }
+            } catch (ClassCastException e) {
+                try {
+                    boolean booleanValue = getBooleanValue(key);
+                    boolean defaultBooleanValue = getDefaultBooleanValue(key);
+                    if (booleanValue != defaultBooleanValue) {
+                        generalPrefs.put(key, booleanValue);
+                    }
+                } catch (ClassCastException e1) {
+                    long longValue = getLongValue(key);
+                    long defaultLongValue = getDefaultLongValue(key);
+                    if (longValue != defaultLongValue) {
+                        generalPrefs.put(key, longValue);
+                    }
+                }
+            }
+        }
+        prefs.put("general", generalPrefs);
+
+        for (String key : ALL_KEYS) {
+            boolean value;
+
+            value = adminSharedPrefs.getBoolean(key, true);
+            if (!value) {
+                adminPrefs.put(key, false);
+            }
+        }
+        prefs.put("admin", adminPrefs);
+
+        return prefs;
+    }
+
+    private String getDefaultStringValue(String key) {
```

Comment:
Earlier no such class was present to hold all default values for the shared preferences. It has been added since it was required for this PR. Also, once we have decided the final structure for the sharedPreferences in #1011 then we can finally replace all the existing uses of shared preferences with the newer approach